up-l1/mqtt_5.adoc

```diff
@@ -0,0 +1,276 @@
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+
+SPDX-FileType: DOCUMENTATION
+SPDX-License-Identifier: Apache-2.0
+----
+
+== Overview
+
+MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). It is designed as an extremely lightweight publish/subscribe messaging transport that is ideal for connecting remote devices with a small code footprint and minimal network bandwidth. MQTT today is used in a wide variety of industries, such as automotive, manufacturing, telecommunications, oil and gas, etc
+
+For more information, please refer to https://mqtt.org/
+
+This document will discuss the uTransport implementation on MQTT 5.
+
+== Specifications
+
+=== UAttribute Mapping
+
+MQTT 5 supports custom header fields, and we leverage this to map the UAttribute values into the MQTT header. Only required uAttribute values are mapped into their own UserProperty fields.
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("1", "{UAttribute.id}") # UUID id
+| UserProperty
+| ("2", "{UAttribute.type}") # UMessageType
+| UserProperty
+| ("3","{UAttribute.source}") # Short Form Serialized Source UURI
+| UserProperty
+| ("4","{UAttribute.sink}") # Short Form Serialized Sink UURI
+| UserProperty
+| ("5","{UAttribute.priority}") # UPriority
+| UserProperty
+| ("9", "{UAttribute.reqId}") # UUID reqid
+| UserProperty
+| ("opt_ua", "{byte array of remaining UAttributes}") # Remaining UAttributes
+|===
+
+All required uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttribute protobuf field number.
+
+The source and sink fields **MUST** be a short form serialized UUri.
+
+Optional uAttributes will be a string representation of a byte array under the UserProperty field with key "opt_ua". This is done to reduce the size of the overall payload. Protobuf serialized byte arrays do not include values in a protobuf message object that are not set, so the byte array will only include the optional UAttributes that are set.
+
+=== URI Mapping to MQTT topics
+
+The MQTT topic a message is published on utilizes elements from both the source and sink UUri fields. Which elements from the UUri are used in the topic depend on whether the sink UUri is present.
+
+Has sink UUri:
+
+`{source.authority.domain}/{source.authority.name}/{sink.authority.domain + ":" + sink.authority.name}/{sink.entity.id}/{sink.entity.version_major}/{sink.resource.id}`
```

Comment:
Is there any added benefit if the domain is in reverse-DNS notation?

Reply:
Removed with update to match topic naming to 1.5.8 UURI spec